sw/device/tests/flash_ctrl_idle_low_power_test.c

```diff
@@ -0,0 +1,164 @@
+
+  uint32_t address = flash_ctrl_testutils_data_region_setup(
+      &flash, kRegionBasePageIndex, kFlashDataRegion, kRegionSize);
+
+  if (rstmgr_reset_info == kDifRstmgrResetInfoPor) {
+    // Create data. Random data will be different than
+    // the 0xFFFFFFFF that is created with an erase.
+    uint32_t data[kNumWords];
+    for (int i = 0; i < kNumWords; ++i) {
+      data[i] = rand_testutils_gen32();
+    }
+
+    // Erasing the page and writing data to it followed
+    // by a read back and compare to sanity check basic operation.
+    CHECK(flash_ctrl_testutils_erase_and_write_page(
+              &flash, address, kPartitionId, data,
+              kDifFlashCtrlPartitionTypeData, kNumWords) == 0);
+    uint32_t readback_data[kNumWords];
+    CHECK(flash_ctrl_testutils_read_page(
+              &flash, address, kPartitionId, readback_data,
+              kDifFlashCtrlPartitionTypeData, kNumWords, 0) == 0);
+    CHECK_BUFFER(data, readback_data, kNumWords);
+
+    // Setting up low power hint and starting watchdog timer followed by
+    // a flash operation (page erase) and WFI. This will create a bite
+    // interrupt at some time following the start of the flash operation.
+    pwrmgr_testutils_enable_low_power(&pwrmgr, kDifPwrmgrWakeupRequestSourceTwo,
+                                      0);
+
+    CHECK_DIF_OK(dif_aon_timer_watchdog_stop(&aon));
+    CHECK_DIF_OK(dif_aon_timer_watchdog_start(&aon, 0x40, 0x80, false, false));
+
+    dif_flash_ctrl_transaction_t transaction = {
+        .byte_address = address,
+        .op = kDifFlashCtrlOpPageErase,
+        .partition_type = kDifFlashCtrlPartitionTypeData,
+        .partition_id = kPartitionId,
+        .word_count = 0x0};
+
+    CHECK_DIF_OK(dif_flash_ctrl_start(&flash, transaction));
+    wait_for_interrupt();
+
+    // Return from interrupt. Check the reset info is still POR and
+    // erase operation completed successfully.
+    CHECK_DIF_OK(dif_rstmgr_reset_info_get(&rstmgr, &rstmgr_reset_info));
+    CHECK(rstmgr_reset_info == kDifRstmgrResetInfoPor);
+
+    CHECK(flash_ctrl_testutils_wait_transaction_end(&flash) == 0);
+
+    CHECK(flash_ctrl_testutils_read_page(
+              &flash, address, kPartitionId, readback_data,
+              kDifFlashCtrlPartitionTypeData, kNumWords, 0) == 0);
+    uint32_t expected_data[kNumWords];
+    memset(expected_data, 0xff, sizeof(expected_data));
+    CHECK_BUFFER(readback_data, expected_data, kNumWords);
+
+    CHECK_DIF_OK(dif_rstmgr_reset_info_clear(&rstmgr));
+  } else if ((rstmgr_reset_info & kDifRstmgrResetInfoLowPowerExit) != 0) {
+    // A boot with low power exit has been detected (not expected.)
+    LOG_ERROR("Reset type shows unexpected low power exit. Reset info = %0x",
+              rstmgr_reset_info);
+    return false;
+  } else {
+    LOG_ERROR("Unexepected reset type detected. Reset info = %0x",
+              rstmgr_reset_info);
+    return false;
+  }
```

Comment:
Since the outcome of both of these else clauses is the same, you could remove lines 152-156. Anything outside of POR reset is an error case.